plugins/plugin-vue/src/script-compilers.js

```diff
@@ -0,0 +1,39 @@
+const esbuild = require('esbuild');
+
+const codeSnippetH = `import { Fragment } from '/web_modules/vue.js';`;
+
+// https://github.com/vitejs/vite/blob/master/src/client/vueJsxCompat.ts
```

Comment:
We have a history with Vite taking code from this repo without proper attribution, so I want to be very careful to not do the same to them. 

Copying code from another open-source project is illegal with proper attribution. If this file is the only one that includes code from the Vite project, then copy-paste their license to the top of this file, like this:

```
/*

This license applies to parts of this file originating from the
https://github.com/vitejs/vite repository:

MIT License

Copyright (c) 2019-present, Yuxi (Evan) You

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

const esbuild = ...
```

If you've also added code from Vite to `plugins/plugin-vue/plugin.js` then add this snippet to a top-level `plugins/plugin-vue/LICENSE` file instead.